Sources/MessagesDisplayDataSource.swift

```diff
@@ -25,41 +25,52 @@
 import Foundation
 
 public protocol MessagesDisplayDataSource: class, MessagesDataSource {
-
-    func messageColor(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> UIColor
-
-    func avatar(for message: MessageType, at indexPath: IndexPath, in messagesCollectionView: MessagesCollectionView) -> Avatar
-
+    
+    func textColor(for message: MessageType, at indexPath: IndexPath) -> UIColor
+    
+    func backgroundColor(for message: MessageType, at  indexPath: IndexPath) -> UIColor
+    
+    func avatar(for message: MessageType) -> Avatar
```

Comment:
I agree in general but I think in respect to the avatar I think it's unnecessary and will not be used thus is just bloat.